### PR TITLE
update to new primitive types from tcgc

### PR DESCRIPTION
**Pull request by iscai-msft**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts [hunk 1]
```diff
@@ -135,10 +135,18 @@ function getCSharpInputTypeKindByIntrinsicModelName(
             return InputPrimitiveTypeKind.SByte;
         case "unit8":
             return InputPrimitiveTypeKind.Byte;
+        case "int16":
```
Comment (timotheeguerin) [inline src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts:138]: missing int8?

Comment (iscai-msft) [reply src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts:138]: int8 is already on line 134

File: src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts [hunk 2]
```diff
@@ -131,14 +131,20 @@ function getCSharpInputTypeKindByIntrinsicModelName(
...
-            return InputPrimitiveTypeKind.Byte;
+        case "int8": // int8 is currently handled as int32 in csharp
+        case "uint8": // int8 is currently handled as int32 in csharp
+        case "int16":
+        case "uint16":
         case "int32":
+        case "uint32":
+        case "numeric":
+        case "integer":
+        case "safeint":
```
Comment (timotheeguerin) [inline src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts:142]: I think safeint also changed in this pr

Comment (timotheeguerin) [reply src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts]: yeah used to be int64

**Pull request closed without merging**